src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -373,6 +400,77 @@ private void runFafReplayFile(Path path) throws IOException, CompressorException
     gameService.runWithReplay(tempSupComReplayFile, replayId, gameType, version, modVersions, simMods, mapName);
   }
 
+  private void hostFromOnlineReplay(int replayId) {
+    downloadReplay(replayId)
+        .thenAccept((path) -> {
+          try {
+            runHostFromFafReplayFile(path);
+          } catch (IOException | CompressorException e) {
+            throw new CompletionException(e);
+          }
+        })
```

Comment:
```suggestion
        .thenCompose((path) -> {
          try {
            return runHostFromFafReplayFile(path);
          } catch (IOException | CompressorException e) {
            throw new CompletionException(e);
          }
        })
```

Reply:
Updated to use Compose approach rather than Accept in latest commit.